test/plugins/webhooks/github.test.js

```diff
@@ -253,6 +253,16 @@ describe('github plugin test', () => {
                     assert.equal(reply.statusCode, 500);
                 });
             });
+
+            it('handles checkouting when given a non-listed user', () => {
+                userFactoryMock.get.resolves(null);
+                userFactoryMock.get.withArgs({ username: 'sd-buildbot' }).resolves(userMock);
+
+                return server.inject(options)
+                .then((response) => {
+                    assert.equal(response.statusCode, 201);
```

Comment:
The stub is guarded in a way that you cannot get a `userMock` if you don't provide `sd-buildbot` as a user: https://github.com/screwdriver-cd/screwdriver/pull/336/files#diff-844b8ba64bd36a7e8d37f9f4d803295dR259